staging/src/k8s.io/apiserver/pkg/storage/etcd3/metrics/metrics.go

```diff
@@ -261,18 +261,29 @@ type monitorCollector struct {
 
 	mutex         sync.Mutex
 	monitorGetter func() ([]Monitor, error)
+	monitors      *[]Monitor
+
+	// get storage metric monitor is protected by mutex
+	monitorMutex sync.Mutex
```

Comment:
I see a mutex above monitorGetter already whose purpose seems similar. Can we reuse that?

Reply:
yea, also this testing PR changed a bit of etcd client behavior. More code refactors may be need if we confirmed this is the right fix. 

currently we are looking for more evidence to build the causality between the gRPC error and the original PR  https://github.com/kubernetes/kubernetes/pull/118812/commits.